inflect/__init__.py

```diff
@@ -2715,6 +2715,14 @@ def _plnoun(  # noqa: C901
         word = Words(word)
 
         if word.last.lower() in pl_sb_uninflected_complete:
+            if len(word.split_) >= 3:
+                for numword in range(1, len(word.split_) - 1):
+                    if word.split_[numword] in pl_prep_list_da:
+                        return " ".join(
+                            word.split_[: numword - 1]
+                            + [self._plnoun(word.split_[numword - 1], 2)]
+                            + word.split_[numword:]
+                        )
```

Comment:
This change adds a lot of complexity to this function, with multiple nested conditionals and loops. This method (`_plnoun`) is already tagged with `noqa: C901` because it's already too complex. Let's not make it more complex. I'd like to explore a way to encapsulate this logic in a method such that it can be documented and compartmentalized.

Reply:
Addressed: https://github.com/jaraco/inflect/pull/188/commits/712906cc85bf0942fd540263965c104936b44f9d
I just can't figure out a way to add this
```
        >>> engine._handle_long_compounds(self, "pair of scissors", 2)
        'pairs of scissors'
        >>> engine._handle_long_compounds(self, "pairs of scissors", 1)
        'pair of scissors'
        >>> engine._handle_long_compounds(self, "pound per square inch", 2)
        'pounds per square inch'
```
in the the `_handle_long_compounds`'s docstring. It expects the `self` argument which we don't pass when we call it in `_plnoun` and `_sinoun`...

Reply:
Try `engine()._handle_long_compounds("pair of scissors", 2)`. Since it's an instance method. Or, you could make it a `@classmethod` and call it without creating an instance (`engine()._handle_long_compounds("pair of scissors", 2)`.

Reply:
Thanks! Addressed here: https://github.com/jaraco/inflect/pull/188/commits/76619bf16a59cb5d3c8e6b687f54ce7fb688fa91